presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -735,3 +778,50 @@ TEST_F(TaskManagerTest, timeoutOutOfOrderRequests) {
           .within(longWait)
           .getVia(eventBase));
 }
+
+TEST_F(TaskManagerTest, aggregationSpilll) {
+  // NOTE: we need to write more than one batches to each file (source split) to
+  // trigger spill.
+  const int numBatchesPerFile = 5;
+  const int numFiles = 5;
+  auto filePaths = makeFilePaths(numFiles);
+  std::vector<std::vector<RowVectorPtr>> batches(numFiles);
+  for (int i = 0; i < numFiles; ++i) {
+    batches[i] = makeVectors(numBatchesPerFile, 1'000);
+  }
+  std::vector<RowVectorPtr> vectors;
+  for (int i = 0; i < filePaths.size(); ++i) {
+    writeToFile(filePaths[i]->path, batches[i]);
+    std::move(
+        batches[i].begin(), batches[i].end(), std::back_inserter(vectors));
+  }
+  duckDbQueryRunner_.createTable("tmp", vectors);
+
+  // Keep bumping up query id per each test iteration to generate a new query
+  // id.
+  int queryId = 0;
+  for (const bool doSpill : {true, true}) {
+    for (const auto& numSpillThreads : {-1, 0, 8}) {
```

Comment:
Does this really test the different number of spilling threads?
spillExecutor() has a static executor inside the function, which initializes once.
The only difference is the run-time check for numSpillThreads <= 0 to return null executor.
Also, what is the point of testing a different number of spilling threads?

Reply:
Yeah, you are right. My previous version is not using static. Will make a change.